crates/driver/src/domain/competition/solution/scoring.rs

```diff
@@ -186,41 +187,65 @@ impl Trade {
         })
     }
 
+    /// The effective amount that left the user's wallet including all fees.
+    ///
+    /// Note how the `executed` amount is used to build actual traded amounts.
+    fn sell_amount(&self) -> Result<eth::TokenAmount, error::Scoring> {
+        Ok(match self.side {
+            order::Side::Sell => self.executed.0,
+            order::Side::Buy => self
+                .executed
+                .0
+                .checked_mul(self.custom_price.buy)
+                .ok_or(Math::Overflow)?
+                .checked_div(self.custom_price.sell)
+                .ok_or(Math::DivisionByZero)?,
+        }
+        .into())
+    }
+
+    /// The effective amount the user received after all fees.
+    ///
+    /// Note how the `executed` amount is used to build actual traded amounts.
+    fn buy_amount(&self) -> Result<eth::TokenAmount, error::Scoring> {
+        Ok(match self.side {
+            order::Side::Sell => self
+                .executed
+                .0
+                .checked_mul(self.custom_price.sell)
+                .ok_or(Math::Overflow)?
+                .checked_div(self.custom_price.buy)
+                .ok_or(Math::DivisionByZero)?,
+            order::Side::Buy => self.executed.0,
+        }
+        .into())
+    }
```

Comment:
Don't we have exactly these methods already in 

https://github.com/cowprotocol/services/blob/2729e8a95eff746d7c19a28a4d5f357df357c15a/crates/driver/src/domain/competition/solution/trade.rs#L150

Reply:
We made `Scoring` module work only with data that is observable onchain, so that it's logic can be reused/shared with autopilot/accounting. More context here: https://github.com/cowprotocol/services/pull/2595#pullrequestreview-1994720357

That said, logic for sell_amount and buy_amount functions is different in `Scoring` => amounts are calculated using custom clearing prices and not uniform clearing prices, as in your attached code. This allows us to not require uniform clearing prices to be part of the settlement calldata anymore (as far as scoring is concerned).